Rubberduck.VBEEditor/Class1.cs

```diff
@@ -0,0 +1,964 @@
+            }
+            _isStart = true;
+        }
+
+        public bool MoveNext()
+        {
+            if (_isStart)
+            {
+                _isStart = false;  // MoveNext is called before accessing the very first item
+                return true;
+            }
+
+            if (_currentTypeLibStruct.Next == IntPtr.Zero) return false;
+
+            _currentTypeLibPtr = _currentTypeLibStruct.Next;
+            _currentTypeLibStruct = StructHelper.ReadStructureSafe<TypeLibObj_VBE>(_currentTypeLibPtr);
+            return true;
+        }
+    }
+
+    // the main class for hooking into the live ITypeLibs provided by the VBE
+    public class TypeLibsAccessor_VBE : DisposableList<TypeLibWrapper_VBE>, IDisposable
+    {
+        public TypeLibsAccessor_VBE(IVBE ide)
+        {
+            // We need at least one project in the VBE.VBProjects collection to be accessible (i.e. unprotected)
+            // in order to get access to the list of loaded project TypeLibs using this method
+
+            foreach (var project in ide.VBProjects)
+            {
+                using (project)
+                {
+                    try
+                    {
+                        using (var references = project.References)
+                        {
+                            // Now we've got the references object, we can read the internal object structure to grab the ITypeLib
+                            var internalReferencesObj = StructHelper.ReadStructure<ReferencesObj_VBE>(references.Target);
+
+                            // Now we've got this one internalReferencesObj.typeLib, we can iterate through ALL loaded project TypeLibs
+                            using (var typeLibIterator = new TypeLibsIterator_VBE(internalReferencesObj.TypeLib))
+                            {
+                                foreach (var typeLib in typeLibIterator)
+                                {
+                                    Add(typeLib);
+                                }
+                            }
+                        }
+
+                        // we only need access to a single VBProject References object to make it work, so we can return now.
+                        return;
+                    }
+                    finally
+                    {
+                        // probably a protected project, just move on to the next project.
+                    }
+                }
+            }
+
+            // return an empty list on error
+        }
+
+        public TypeLibWrapper_VBE FindTypeLib(string searchLibName)
+        {
+            foreach (var typeLib in this)
+            {
+                if (typeLib.Name == searchLibName)
+                {
+                    return typeLib;
+                }
+            }
+
+            throw new ArgumentException($"TypeLibsAccessor_VBE::FindTypeLib failed. '{searchLibName}' project not found.");
```

Comment:
should definitely `return null;`. If searching for a value fails, I don't expect a `List` to give me an `ArgumentException`